Design Picker: Add design preview

arthur791004 opened the pull request:
#### Changes proposed in this Pull Request
 
* Implement `DesignButtonCover` to show the `Start` and `Preview` buttons when you hover on the design so that we can choose different options.
* Use `stepSectionName` to decide the selected design and then we can preview and back easliy.
* Use `/template/demo` endpoint to show the preview with translation. The only problem I have no idea is that when I use this endpoint as previewUrl for `<WebPreview />` component, it will misunderstand the loading result as failure and redirect the whole page. Therefore, I use `disableRedirects` to fix this problem temporarily
* Use next button base on https://github.com/Automattic/wp-calypso/pull/56311 to show `Start with <theme>` button on the top-right corner.
* Make `<WebPreview />` component accept `toolbarComponent` so that we can customize the toolbar we need
* Don't show `DesignButtonCover`  on mobile because it cannot preview different device

##### Demo
https://user-images.githubusercontent.com/13596067/133767836-3f3536ae-bab2-41d3-8a98-a97a29b26e0f.mov

##### Preview with translation

![image](https://user-images.githubusercontent.com/13596067/134478619-e28c56fc-4458-45d9-ad51-b86ee6459ffd.png)


#### Dev Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* ~Apply D67443-code and follow its TEST PLAN~ (**Deployed**), but ensure pointing your API to production because iframe preview doesn't work on the sandbox. See: p1632891540262700-slack-C029SB8JT8S
* Go to `/start/setup-site?siteSlug=<your_site>`
* Hover on the design
* Select `Preview <design_name>`
* Choose different devices
* Click `Start with <design_name>` button at top-right corner

#### User Facing (non-dev)  Testing instructions
* Go to `https://container-vigilant-gould.calypso.live/start/domains?flags=signup/setup-site-after-checkout`
* Go through the signup flow and get to the Design Picker
* Hover on any design
* Select `Preview <design_name>`
* Choose different devices
* Click `Start with <design_name>` button at top-right corner

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/56302

arthur791004 committed:
Design Picker: Add preview section

arthur791004 committed:
Add DesignButtonCover for showing start or preview on hover

arthur791004 committed:
Fix incorrect design title

arthur791004 committed:
Query selected theme and use demoUrl for preview

roo2 commented:
hmm there's a bit of a usability issue with the navigation buttons from the preview page
If you've scrolled down on the theme picker page, the theme preview page will start scrolled down and you won't be able to see the back and `Start with <theme>` buttons. This is made worse because the preview window can capture scroll.

Potential solutions I can think of:
Using css `position: fixed` on the top nav so that buttons are always visible
Reseting scroll to 0 when entering the preview page.
Using a modal instead of a seperate page.

![issue](https://user-images.githubusercontent.com/22446385/134107267-9f0620a1-54f0-4739-b704-133a8bb1b600.gif)

roo2 commented on client/signup/step-wrapper/style.scss:
```diff
@@ -92,3 +92,19 @@
 	}
 }
+
+.step-wrapper__navigation-links {
```
It looks like the `step-wrapper__navigation-links` class doesn't exist in the jsx as far as I can see?

arthur791004 replied:
Oops...I forgot to remove this

roo2 commented on client/signup/steps/design-picker/style.scss:
```diff
@@ -58,3 +58,22 @@
 }
+
+.design-picker__preview {
+	height: 860px;
```
Setting height here is what's causing the scrolling issue.

arthur791004 commented:
> If you've scrolled down on the theme picker page, the theme preview page will start scrolled down and you won't be able to see the back and Start with <theme> buttons. This is made worse because the preview window can capture scroll.

@roo2 Good catch 👍  Let me think how to deal with it better!

arthur791004 committed:
Move up the header and content for design preview

arthur791004 committed:
Support design preview on mobile

arthur791004 committed:
Fix scroll position problem

arthur791004 commented:
@roo2 I've fixed the scroll position problem by
1. Keep scroll position and reset scroll top to 0 when entering preview
2. Restore scroll position when leaving preview

> I think that the right fix is to adjust the page layout so that the total page height is no more than 100% of the viewport and scrolling isn't neccessary.
> Using css position: fixed on the top nav so that buttons are always visible

The problem for these two is that when the user go back from preview, the previous scroll position will lose. They cannot see the previous selected design immediately.

> Using a modal instead of a seperate page.

The problem is that the wordpress icon on the left side of back button might not show on the modal.

arthur791004 committed:
Change device icon

arthur791004 committed:
Adjust height of iframe

arthur791004 committed:
Get previewUrl by getDesignUrl so that we can have live preview with translation

roo2 commented:
This is behaving really nicely now, I'm just looking through the code 👀

roo2 commented:
one interesting issue, probably not related to this PR is that if the page is reloaded, calypso loses track of which `siteSlug` was used originally and we get a javascript error. It would be nice if we can support using refresh while on this page without losing your place in the flow. I don't think we need to deal with this in this PR but just bringing it up.
<img width="643" alt="Screen Shot 2021-09-24 at 10 46 51 am" src="https://user-images.githubusercontent.com/22446385/134601830-860af264-65b1-4317-a358-e37645d73946.png">

arthur791004 commented:
@roo2 Yep, I also found this existed problem. If we go back from browser, we'll lose the dependencies from query string and validation will fail.

roo2 commented on client/components/web-preview/content.jsx:
```diff
@@ -232,7 +231,7 @@ export class WebPreviewContent extends Component {
 			debug( 'preview loaded for url:', this.state.iframeUrl );
 		}
-		if ( this.checkForIframeLoadFailure( caller ) ) {
+		if ( ! this.props.disableRedirects && this.checkForIframeLoadFailure( caller ) ) {
```
I looked into why checkForIframeLoadFailure is returning true.

We are counting on the content in the iframe to send a `postMessage` to calypso with the `partially-loaded` message. In other iframe previews (e.g. a post preview), we load https://s0.wp.com/wp-content/mu-plugins/jetpack-iframe-embed/jetpack-iframe-embed.js which is responsible for sending the `partially-loaded` message.

I think we should work out why jetpack-iframe-embed.js is not being loaded by the iframed content, because there might be other things that jetpack-iframe-embed.js is doing which we also want to happen. (although it seems to work without it)

arthur791004 replied:
Cool! I don't notice we have `jetpack-iframe-embed.js`. I think the problem is that I use the API endpoint as the iframe URL directly, so it won't have that script.

If we don't want `disableRedirects `, I think one solution might be that fetch demo site from API  first and then use the response as `previewMarkup`. Or we can just use iframe directly rather than `<WebPreview />`

roo2 replied:
Hmm this feels over my head technically. Maybe @noahtallen or @jsnajdr would have an opinion? I believe that you both have quite a deep understanding of this calypso iframe.

What's happening: 

Here we are loading a preview of a demo template loaded from the public api directly in the preview iFrame e.g.
`https://public-api.wordpress.com/rest/v1/template/demo/barnsbury/barnsbury`
It looks like in other places where we use the iframe preview, we wrap the content in a way which (among other things) inserts the `https://s0.wp.com/wp-content/mu-plugins/jetpack-iframe-embed/jetpack-iframe-embed.js` script into the page loaded inside the iframe. This script is what notifies the WepPreview that the iframe'd content has loaded correctly via a postMessage.

This implementation of loading the template demo content directly seems to work well with this one logic modification. ( which basically removes the `checkForIframeLoadFailure` check )

The only issue I see is that we're not really checking for the iframe failing to load. However, this is probably much less likely to occur when loading our own template previews, versus when loading the users custom content. Another issue might be lack of consistency with other ways that we implement preview.

@noahtallen or @jsnajdr Do you think that this approach is alright for our situation? 🙏 

Thank you
My brain hurts.

jsnajdr replied:
The `WebPreview` component is designed to embed documents that have the `jetpack-iframe-embed` script and are able to talk to the parent frame using the specified message protocol. If you don't want that, you can load the document into your own `iframe`, without using `WebPreview` at all. Or does `WebPreview` have any functionality of does it provide any services that you really need here?

Looking at the `/template/demo` REST endpoint, I see that it pipes through a rendered page of that particular theme's demo site. So it actually _is_ a preview of a WordPress site and it would be nice to have the `jetpack-iframe-embed` script active it it. To achieve that, you need to have some query args in the site URL, like:
```
https://rockfieldstarter.wordpress.com?iframe=true&theme_preview=true&frame-nonce=...
```
The `iframe` query arg, together with either `preview` or `theme_preview` query arg, is what triggers the `jetpack-iframe-embed` script. See the `Jetpack_Iframe_Embed::is_embedding_in_iframe` PHP method.

To pass through the `iframe` query arg, you'll need to modify the following statement in the `WPCOM_JSON_API_Template_Demo_Endpoint` PHP class:
```php
$demo_content_response = $template_helpers->get_demo_content(
  $theme_demo_site_id,
  [
    'theme_preview' => 'true',
    'language'      => $language,
  ],
);
```
and pass through also the `iframe` query arg of the REST endpoint, just like it already passes through the `language` param.

You'll probably also need to work with the `iframe-nonce` param, but I don't know the details.

Doing all this, you should arrive at perfect theme demo site previews 🙂

arthur791004 replied:
Interesting! I've modified `WPCOM_JSON_API_Template_Demo_Endpoint` PHP class and let it support `iframe` query arg here: D67443-code

roo2 commented on client/signup/steps/design-picker/icons.jsx:
```diff
@@ -0,0 +1,46 @@
+import { Path, SVG, Rect } from '@wordpress/components';
+import React from 'react';
+
+export const computer = (
```
there are some existing device icons used in the device preview for individual posts. I think that we should be consistent, if we change to these ones we should update the other previews. e.g. on the  `pages/` screen, when clicking the view icon next to a page
<img width="904" alt="Screen Shot 2021-09-24 at 5 03 01 pm" src="https://user-images.githubusercontent.com/22446385/134632371-3d349c03-0510-4fc3-a9a9-9e6e8522e853.png">

arthur791004 replied:
But if we go to `/new/style`, you'll find that the icon is the new one 😅

arthur791004 replied:
![image](https://user-images.githubusercontent.com/13596067/134649307-5dcbc3a9-7a54-45ff-99d7-70693122d6c9.png)

roo2 replied:
ahh ok interesting. I see that it's not so easy to reuse the assets because they're from gutenboarding which is a separate app (which may not be maintained long term). There was a `// fixme` comment where the icons where originally used in gutenboarding: 
```
// TODO FIXME: React elements are a poor choice for static svgs.
// Refactor, external svg with use?
```
I also think external SVGs are better because they are the source format and compatible with external editors etc. (but I think it's fine here for now)

✅

roo2 replied:
I think a future improvement would be to update the other icons used on the other preview windows in calypso to use these icons, but it's good for this PR 👍

arthur791004 replied:
Cool! I don't notice that there was a `// fixme` comment there!

roo2 commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -26,7 +36,54 @@ class DesignPickerStep extends Component {
+	updateScrollPosition() {
+		if ( this.props.stepSectionName ) {
+			this.setState( { scrollTop: document.scrollingElement.scrollTop } );
+			document.scrollingElement.scrollTop = 0;
```
It looks to me like you've found a CSS solution for the height of this page as well, so it looks like the javascript scrolling fix is no longer neccessary.

arthur791004 replied:
Maybe we should keep this for safety if the height still makes vertical overflow in a certain device or certain window size?

roo2 replied:
I think that this would only be necessary if a browser didn't support the `vh` unit or the css `calc` `hieght: calc(100vh - 256px);`. Since we're not supporting IE11 anymore p4TIVU-9Ir-p2 I'm confident that the css fix will be enough. I went and tested safari and edge and it's looking good on them 👍

arthur791004 replied:
Okay, then I'll remove it

roo2 reviewed (commented):
Nice work fixing the scrolling issue! I have some other questions/feedback

roo2 commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -26,7 +36,54 @@ class DesignPickerStep extends Component {
+		const { stepSectionName } = this.props;
+		const { availableDesigns } = this.state;
+
+		this.setState( {
```
it's a shame we didn't make this step a functional component originally 🤷 

👍

roo2 commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -26,7 +36,54 @@ class DesignPickerStep extends Component {
+	};
+
+	previewDesign = ( selectedDesign ) => {
+		page( getStepUrl( this.props.flowName, this.props.stepName, selectedDesign.theme ) );
```
Prehaps this is where we'd look to enable reloading the page without using the `siteSlug` context.

roo2 commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -63,13 +115,43 @@ class DesignPickerStep extends Component {
+				showEdit={ false }
+				externalUrl={ siteSlug }
+				previewUrl={ previewUrl }
+				loadingMessage={ translate( '{{strong}}One moment, please…{{/strong}} loading your site.', {
```
I think we would usually use more succinct language, what do you think @ollierozdarz 

I'd probably just go with:

> loading your site

arthur791004 replied:
I just re-use the wording from https://github.com/Automattic/wp-calypso/blob/f4b8378eaa8772d0f7dc1ef2fd2c20e7741adf52/client/my-sites/preview/main.jsx#L182 😅

roo2 replied:
Ahh I missed that, sorry. Ok that's fine then, or at least not an issue with this PR, I still think we'd usually be more succinct but 🤷 👍.

p-jackson commented:
Looks like the unit tests are failing because of the added `useViewportMatch` hook, and JSDOM doesn't implement the `window.matchMedia`.

We could mock the `useViewportMatch` hook, but it's in a test called `integration.test.tsx` so it feels like we should make it as integration-y as possible.

I added this to `packages/calypso-build/jest/setup.js`
```js
// Attempt not to define matchMedia if Jest isn't running in JSDOM environment
if ( typeof window !== 'undefined' ) {
	// https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
	Object.defineProperty( window, 'matchMedia', {
		writable: true,
		value: jest.fn().mockImplementation( ( query ) => ( {
			matches: false,
			media: query,
			onchange: null,
			addListener: jest.fn(), // deprecated
			removeListener: jest.fn(), // deprecated
			addEventListener: jest.fn(),
			removeEventListener: jest.fn(),
			dispatchEvent: jest.fn(),
		} ) ),
	} );
}
```

This adds the mock to every single test runner so maybe it's overkill. It'd also be possible to edit `packages/design-picker/jest.config.js` and add it only to the design picker's test suite.

p-jackson commented:
Just had a play with this and seeing the full view of the site is really cool!

Have you got any thoughts on how this works with keyboard navigation? I noticed the buttons are still receiving focus even when they're invisible. Would it make sense for the buttons to appear and disappear as you tab through each of the themes? So the buttons are visible if the focus is anywhere within the the thumbnail?

I think what's happening on touch devices is that it assumes you want to preview? I also wondered whether it might feel better to do a one tap to display the buttons, and then a second tap to choose the option. Without the context of seeing the button text "Preview <theme>" I don't think it's as obvious that what you're looking at is just a preview. Maybe I think this is my actual site. Should I be able to click now to edit in a wysiwyg fashion?

jsnajdr commented:
> unit tests are failing because of the added useViewportMatch hook, and JSDOM doesn't implement the window.matchMedia.

We're usually mocking `window.matchMedia` directly in the test file when we need it. I found 5+ instances of this in the Calypso repo, one example is here: https://github.com/Automattic/wp-calypso/blob/trunk/packages/viewport/test/index.js#L27

arthur791004 commented:
> I think what's happening on touch devices is that it assumes you want to preview? I also wondered whether it might feel better to do a one tap to display the buttons, and then a second tap to choose the option. 

@p-jackson I also discussed this with @ollierozdarz and he said "Yeah I think for a quick v1 push, that’s the smartest. We could put the same buttons below each design on mobile, but we can always do that later if we see it’s an issue with users (which I don’t feel like it will be)". Also, I agreed with his point and don't feel like it will be either. So, we can do this enhancement later according to more feedbacks.

> Without the context of seeing the button text "Preview " I don't think it's as obvious that what you're looking at is just a preview. Maybe I think this is my actual site. Should I be able to click now to edit in a wysiwyg fashion?

I think this situation might also happen in https://wordpress.com/new/style and it seems to work well now?

arthur791004 committed:
Add default mock functions for window.matchMedia

roo2 commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -82,23 +164,52 @@ class DesignPickerStep extends Component {
-export default connect( null, { submitSignupStep } )( localize( DesignPickerStep ) );
+export default compose(
+	connect( null, { submitSignupStep } ),
+	withViewportMatch( { isMobile: '< small' } ),
```
I think that `useMobileBreakpoint()` from `@automattic/viewport-react` is a preferred way to do this.

It looks like this pattern was coppied from `/gutenboarding/`, where we don't have access to all of the same libraries that we do in the start flow.

arthur791004 replied:
As I know, `@wordpress` packages are more powerful, so I choose to use them. Another reason is that `@automattic/viewport-react` doesn't have the type definition and it leads the package building to fail 😂

p-jackson replied:
I think preferring the `@wordpress/` component helpers makes sense. We've started to use the gutenberg breakpoint scss mixins and deprecated the calypso ones as another example. I also like type definitions :)

arthur791004 replied:
👍

roo2 replied:
ok, that makes sense to prefer `@wordpress/*` in general. 

My concern with it is that whereas the `viewport-react` component has a simple interface of `isMobile`, and `isDesktop`, the more powerful `@wordpress` withViewportMatch  requires you to choose a definition for mobile and desktop, so there's a possibility that `isMobile` and especially `isDesktop` would be defined differently in different places. e.g. should `isDesktop` be defined as `> mobile` or `> large`?

But ok, that's just something to think about, I understand/agree it makes sense to use it here, especially as the other packages don't have types 👍

roo2 commented:
hmm on mobile, it is currently always showing the preview. I think it makes more sense to not show it at all and just select the design in one click since as you said, the device previews don't work on mobile

roo2 commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -82,23 +164,52 @@ class DesignPickerStep extends Component {
+					fallbackSubHeaderText={ '' }
+					subHeaderText={ '' }
+					stepContent={ this.renderDesignPreview() }
+					align={ isMobile ? 'left' : 'center' }
```
I'm not sure what effect `align={ isMobile ? 'left' : 'center' }` is having on mobile, I suspect it isn't neccessary to test for mobile here.

arthur791004 replied:
According to the EuqfSnWpfYx8fgiBlVmbuA-fi-2497%3A1481, we have to make the title align left on mobile. Here is the screenshot
![image](https://user-images.githubusercontent.com/13596067/134897497-b01ad5ed-5181-4ac4-8e35-419bd1596c81.png)

roo2 replied:
ahh ok 👍 sorry I missed that

roo2 commented on client/signup/steps/design-picker/preview-toolbar.scss:
```diff
@@ -0,0 +1,98 @@
+}
+
+// Need the extra specificity here to override core
+.preview-toolbar__button.preview-toolbar__button {
```
this is a neat trick that I haven't seen before 👍 😂

jsnajdr replied:
A preferred way to create the extra specificity would be to repeat the Core classes that are being overridden, something like:
```
.components-button.preview-toolbar__button
```
That makes is much more clear which Core class specifically is being overridden.

arthur791004 replied:
Just select itself twice to increase the CSS priority 😂

arthur791004 replied:
Both ways are okay with me. So I changed to
```css
.button.preview-toolbar__button
```

roo2 commented on packages/design-picker/src/components/index.tsx:
```diff
@@ -102,9 +104,80 @@ const DesignButton: React.FC< DesignButtonProps > = ( {
+	onPreview,
+	...props
+} ) => {
+	const isDesktop = useViewportMatch( 'large' );
```
here too I think we should be using `useDesktopBreakpoint` from `viewport-react`

roo2 commented:
One small visual thing, on safari the buttons aren't spaced as we'd expect
<img width="704" alt="Screen Shot 2021-09-27 at 8 35 51 pm" src="https://user-images.githubusercontent.com/22446385/134893018-9d5265cc-ee8c-4c23-a12c-612aeda1bdb0.png">

roo2 reviewed (commented):
OK, I've looked through everything, thanks for your patience with all of the feedback

arthur791004 committed:
Add iframe args to getDesignUrl so that it can be used by WebPreview component

arthur791004 commented:
@roo2 After consideration, I think we still preview on mobile because our users can see what it looks like on mobile even if it cannot change the device.

arthur791004 committed:
Improve a11y for DesignButtonCover

arthur791004 commented:
> One small visual thing, on safari the buttons aren't spaced as we'd expect
> <img alt="Screen Shot 2021-09-27 at 8 35 51 pm" width="704" src="https://user-images.githubusercontent.com/22446385/134893018-9d5265cc-ee8c-4c23-a12c-612aeda1bdb0.png">

What's your safari version? My safari (Version 14.1.1 (16611.2.7.1.4)) works well 😂
![image](https://user-images.githubusercontent.com/13596067/134910266-96688cc6-b9e6-442d-b4a1-131a24d5ecd0.png)

arthur791004 committed:
Some changes according to review comment

p-jackson commented:
> > ... to do a one tap to display the buttons, and then a second tap to choose the option.
>
> we can always do that later if we see it’s an issue with users

Fair enough.

> I think we still need the preview on mobile because our users can see what it looks like on mobile even if they cannot change the device.

FWIW I also think we should have the site preview on mobile too.

The keyboard nav changes are great!

https://user-images.githubusercontent.com/1500769/134992904-dfac4acb-b84b-44ba-b729-330264abfb6f.mov

After pulling the latest changes I found that the page redirected to the iframe src url after the preview finished loading. Can be seen at the end of the above video. I think it happens whether you go to the preview with mouse or keyboard.

roo2 commented:
> What's your safari version? My safari (Version 14.1.1 (16611.2.7.1.4)) works well 😂

Ahh mine is 13.1.3 (15609.4.1)

Ok, since it's minor and only affects older versions, I think that's fine 👍

arthur791004 committed:
Change to use margin-top rather than gap to support safari 13.1

arthur791004 commented:
> > What's your safari version? My safari (Version 14.1.1 (16611.2.7.1.4)) works well 😂
> 
> Ahh mine is 13.1.3 (15609.4.1)
> 
> Ok, since it's minor and only affects older versions, I think that's fine 👍

The problem is I using `gap: 8px` but it's not supported by safari 13.1. I'll fix this one by changing to use `margin-top`

roo2 commented:
ok, this is looking good to me, great work getting D67443-code up so fast, I'm glad we got that feedback from Jarda and you were able to incorporate it so quickly. Nice work 😄

roo2 reviewed (approved):
Good job implementing the new iframe behavior in D67443-code. I was just wondering what end user benefit we get from that change. Now if for some reason the content fails to load in the iframe, the browser will redirect to the template demo source site directly. (which I assume is more likely to work?)

I think the main thing we get is consistency with how the `WebPreview` `iframe` works and not having to add that special `ignoreRedirect` condition.

Nice work anyway!

arthur791004 committed:
Fix start with design not work

jsnajdr commented:
> I was just wondering what end user benefit we get from that change.

@roo2 There are several little but nice services provided by the `jetpack-iframe-embed` script and its cooperation with the parent frame:
- links that lead outside the site, e.g., the `https://rockfieldstarter.wordpress.com/` demo has a "Create a free website or blog at WordPress.com" to wordpress.com in the footer, will be opened in a new tab instead of inside the iframe. The iframe navigates only inside the previewed site, never outside.
- if the frame content fails to load, most likely because of third-party cookie-blocking issues, the parent frame will be able to detect that and open in a new top-level tab instead.
- the embedded iframe listens for events like Esc keypress or focus and notify the parent frame to do things like close the preview modal or close some popup menu when something else is focused.
- when the iframe loads a page (initially or on navigation), it provides the parent frame with some detailed info about progress, letting the parent frame to show a nice loading indicator.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6678196

Thank you @arthur791004 for including a screenshot in the description! This is really helpful for our translators.

arthur791004 committed:
Fix Code style

kristastevens commented:
Test Flow: 1. Signup w/ free plan

I previewed all theme designs across all the devices. Everything looks smooth and runs quickly. I did notice that the **tablet layouts for Laporte and Coutoire** seemed a little off:

<img width="868" alt="Laporte — tablet view" src="https://user-images.githubusercontent.com/4358882/135914154-41f8559c-a438-4d1c-8c99-ced7c34e13c5.png">

<img width="868" alt="Coutoire — tablet view" src="https://user-images.githubusercontent.com/4358882/135914204-336a967f-8ea9-4ab0-a4e7-5f9abfd4cfb8.png">

arthur791004 commented:
@kristastevens Thank you for checking all theme designs across all the devices and notice some layouts look strange! But I think the tablet layout for those two seemed a little off for a while 😅  If you go to `My Home > Appearance > Themes > Select Laporte or Coutoire > Open live demo` and switch the device to tablet, it would be the same.

![image](https://user-images.githubusercontent.com/13596067/135975171-6cac3b78-94e5-425c-9d2d-9cad06fb86b8.png)

![image](https://user-images.githubusercontent.com/13596067/135975620-5ee2dbce-461f-4ed6-858e-c5ea3c053b53.png)

autumnfjeld commented:
## Testing Flow 4 from pdgK6S-c4-p2
✅ 

* Starting url: (`/start/free`) https://container-funny-greider.calypso.live/start/free/domains?flags=signup/setup-site-after-checkout
* Chose the personal plan
* Paid with store admin credits
* I get this congratulations screen
* ![image](https://user-images.githubusercontent.com/5665959/136080327-751dbbfd-2700-4c25-b12e-1c71a0328bc1.png)
* Then I clicked the button at the bottom (I think it says see my plan or something like that) and I am taken to /plans/my-plan
* ⛔  Never saw design Picker. I believe this is part of the know flow issues reported here https://github.com/Automattic/wp-calypso/issues/56303  **So not a blocker for this PR.**

## Testing Flow 3 from pdgK6S-c4-p2
✅  All good.
One small design issue, not a blocker for this PR:
![image](https://user-images.githubusercontent.com/5665959/136080805-4ff711fc-ef45-4183-97f5-d651e82a55dd.png)
In comparison, the mobile preview looks correct on /theme/bantry{site}

arthur791004 commented:
I file a issue for the tablet layout of those two themes: https://github.com/Automattic/wp-calypso/issues/56811

Greatdane commented:
2. Signup w/  with paid plan (https://container-nostalgic-panini.calypso.live/)

More of a design thing, but when choosing a plan, there is no 'on hover' event for the Personal/Business/eCommerce plan. (There is an 'on hover' event for all items on the domains page)
<img width="1454" alt="Markup 2021-10-06 at 17 24 21" src="https://user-images.githubusercontent.com/5075840/136168753-25d0b72f-e5fd-4c41-be58-9bf01a4ff376.png">

When I choose a plan, I can't go back after the checkout page. Clicking [X] on the checkout page directs me to the plans page. Is this the intended workflow? On the current live site, we get directed to the home page to edit.
Going back to the checkout, I then got the congratulations page, **but never got to the design picker**. ⛔

<img width="969" alt="Markup 2021-10-06 at 17 30 39" src="https://user-images.githubusercontent.com/5075840/136169191-ef257629-2913-4cbe-bec7-5360029338e3.png">

I believe this is noted here; https://github.com/Automattic/wp-calypso/issues/56303

Will test more tomorrow.

jromales commented:
Testing Flow 2 from pdgK6S-c4-p2
✅  
- Starting url: https://container-wizardly-carver.calypso.live/start/domains?flags=signup/setup-site-after-checkout
- Chose eCommerce (monthly) plan and paid with store admin credits
- Then I get this screen:
![Screen Shot 2021-10-06 at 11 01 41 PM](https://user-images.githubusercontent.com/89536719/136231908-5c249f59-f477-4657-9149-651ad580def2.png)
- Next, I click on `Create your store!` and I'm taken to this screen: 
![Screen Shot 2021-10-06 at 11 01 56 PM](https://user-images.githubusercontent.com/89536719/136232403-98ad337a-f5b5-441f-9eaf-13044a994856.png)
- I click `Skip setup store details` and I'm taken to this screen at /wp-admin/admin.php?page=wc-admin:
![Screen Shot 2021-10-06 at 11 20 20 PM](https://user-images.githubusercontent.com/89536719/136233439-bdbb0206-d65a-409f-ac39-af3947f56176.png)
- I never saw the `Design Picker` (as @autumnfjeld mentioned, this is not a blocker for this PR.)
